src/plugins/ui_actions/public/service/ui_actions_service.ts

```diff
@@ -123,22 +123,17 @@ export class UiActionsService {
    * `addTriggerAction` is similar to `attachAction` as it attaches action to a
    * trigger, but it also registers the action, if it has not been registered, yet.
    */
-  public readonly addTriggerAction = (
-    triggerId: string,
-    action: ActionDefinition // TODO: remove `Action` https://github.com/elastic/kibana/issues/74501
-  ): void => {
+  public readonly addTriggerAction = (triggerId: string, action: ActionDefinition<any>): void => {
     if (!this.actions.has(action.id)) this.registerAction(action);
     this.attachAction(triggerId, action.id);
   };
 
-  public readonly getAction = <T extends ActionDefinition>(
-    id: string
-  ): Action<ActionContext<T>> => {
+  public readonly getAction = (id: string) => {
```

Comment:
Maybe we should downcast it to Action ?

```suggestion
  public readonly getAction = (id: string): Action => {
```

Reply:
Done.